### PR TITLE
sci-libs/metis-5.1.0-r1: bad pkg-config file is generated (#588148 by…

### PR DESCRIPTION
**Pull request by g-braeunlich**
… Viacheslav Ostroukh)

### CODE REVIEW

File: sci-libs/metis/metis-5.1.0-r1.ebuild [hunk 1]
```diff
@@ -11,7 +11,7 @@ HOMEPAGE="http://www-users.cs.umn.edu/~karypis/metis/metis/"
 SRC_URI="http://glaros.dtc.umn.edu/gkhome/fetch/sw/${PN}/${P}.tar.gz"
 
 SLOT="0"
-KEYWORDS="~amd64 ~arm ~hppa ~ppc ~ppc64 ~sparc ~x86 ~amd64-fbsd ~x86-fbsd ~amd64-linux ~x86-linux"
+KEYWORDS="~alpha ~amd64 ~arm ~hppa ~mips ~ppc ~ppc64 ~sparc ~x86 ~amd64-fbsd ~x86-fbsd ~amd64-linux ~x86-linux"
```
Comment (SoapZA) [inline sci-libs/metis/metis-5.1.0-r1.ebuild]: have you tested the alpha and mips arches? These are "specialty" type architectures, for which you usually have to be on the corresponding team to be able to keyword them.

Comment (SoapGentoo) [inline sci-libs/metis/metis-5.1.0-r1.ebuild]: Keywording non-x86 and non-amd64 arches requires permission from the corresponding teams. Better remove these.

File: sci-libs/metis/metis-5.1.0-r1.ebuild [hunk 2]
```diff
@@ -55,16 +55,16 @@ src_test() {
 
 src_install() {
 	cat >> "${T}"/metis.pc <<- EOF
-	prefix="${EPREFIX}"/usr
-	exec_prefix="${EPREFIX}"/usr
-	libdir="${EPREFIX}"/usr/$(get_libdir)
-	includedir="${EPREFIX}"/usr/include
+	prefix=${EPREFIX}/usr
+	exec_prefix=\${prefix}
+	libdir=\${prefix}/$(get_libdir)
```
Comment (SoapGentoo) [inline sci-libs/metis/metis-5.1.0-r1.ebuild]: While in this case it doesn't make a big difference, pkg-config files have a canonical layout, see
https://people.freedesktop.org/~dbn/pkg-config-guide.html

i.e. change `libdir=\${prefix}/$(get_libdir)` to `libdir=\${exec_prefix}/$(get_libdir)`

And maybe swap `libdir` and `includedir` around.

File: sci-libs/metis/metis-5.1.0-r1.ebuild [hunk 3]
```diff
@@ -55,16 +55,16 @@ src_test() {
...
+	includedir=\${prefix}/include
 
 	Name: METIS
 	Description: Software for partioning unstructured graphes and meshes
 	Version: ${PV}
-	Libs: -L${libdir} -lmetis
-	Cflags: -I${includedir}/metis
+	Libs: -L\${libdir} -lmetis
+	Cflags: -I\${includedir}/metis
```
Comment (SoapGentoo) [inline sci-libs/metis/metis-5.1.0-r1.ebuild]: Good, just swap them around.

File: sci-libs/metis/metis-5.1.0-r1.ebuild (hunk 1 above)
Comment (g-braeunlich) [inline sci-libs/metis/metis-5.1.0-r1.ebuild]: OK. Fixed.

**Comment by SoapGentoo** [comment]
@g-braeunlich thanks 😄  good for a first commit.
Change the **COMMITMSG** to something like "`sci-libs/metis: Produce standard pkg-config file`"
followed by an empty line, and then
`Gentoo-bug: 588148`
All of this is made much easier if you use `repoman commit` instead of `git commit`

For the standard Gentoo git workflow, see:
https://wiki.gentoo.org/wiki/Gentoo_git_workflow

Also, the GLEP being drafted:
https://wiki.gentoo.org/wiki/User:MGorny/GLEP:66

And finally, git's most powerful tool: git **rebase**, which allows you to rewrite the history graph, i.e. combining multiple commits into one, reordering commits, etc:
http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html
Always combine (=squash in git-speak) commits into logical changes.

**Comment by g-braeunlich** [comment]
How can I push my rebased commit? Currently I have two commits. In between these two commits there are other commits fetched from upstream. So I tried to delete both of my two commits, then
git pull --rebase=preserve
and finally commit and push the total changes.

**Comment by SoapGentoo** [comment]
@g-braeunlich as git rebase changes the history, you have to force push with `git push -f`

**Comment by g-braeunlich** [comment]
OK. Thanks. Next problem: using repoman, the commits get signed.
In this case, git push --signed -f, states "fatal: the receiving end does not support --signed push". How can I handle this?

**Comment by SoapGentoo** [comment]
There's a difference between signing commits and a signed git push. Github **doesn't support signed pushes**, it has no problem with signed commits though. Hence, just do `git push -f` (without the `--signed` part).

**Comment by g-braeunlich** [comment]
OK. Had to pass --signed=no

**Comment by g-braeunlich** [comment]
Ah: and how do I honor the patch, the bug owner supplied?

**Comment by SoapGentoo** [comment]
You can add as much commentary as you like after the `Gentoo-bug:` line. You can either use `git commit --amend` to edit your last **COMMITMSG**, or use `git reset --soft`do undo the last commit (and thereby dropping the files back into the staging area) and then doing `repoman commit` again.

**Comment by SoapGentoo** [comment]
@g-braeunlich Could you also port the ebuild to `EAPI=6`? Seeing it's an ebuild with unstable keywords, this is not a problem. Also, you need to revbump the ebuild, as you are changing files on disk that affect users.

**Comment by g-braeunlich** [comment]
Does a revbump involve more than just adding a renamed ebuild?

**Comment by SoapGentoo** [comment]
No, `git mv metis-5.1.0-r{1,2}.ebuild` should be enough. Have you tested the new ebuild and installed it?

**Comment by g-braeunlich** [comment]
OK. Just merged the ebuild for test. Runs fine.

**Comment by SoapGentoo** [comment]
@g-braeunlich Thanks for the PR, you're off to a good start. Could you maybe try to modify the buildsystem in such a way (don't worry, not in the repo) in order to integrate native pkg-config support and submit it upstream? CMake doesn't natively support generating pkg-config files, but if you have a look how x265 and SeqAn do it, you can see that it's not hard. I believe this is what we should aim for as good citizens in the free and open-source software community, so that other distros can also profit. See for instance how I got pkg-config into SeqAn: https://github.com/seqan/seqan/issues/1042

**Comment by g-braeunlich** [comment]
Ok, I will have a look at this.
Could you close the corresponding bug: https://bugs.gentoo.org/show_bug.cgi?id=588148 ?

**Pull request closed without merging**